R/render-plot.R

```diff
@@ -570,6 +570,17 @@ find_panel_info_api <- function(b) {
       domain$bottom <- -domain$bottom
     }
 
+    # Remember the x/y range if it's a faceted
+    # plot with a free discrete axis. This is necessary
+    # to properly inverse map the numeric (i.e., trained)
+    # positions back to the data scale (#2410)
+    if (nrow(layout) > 1) {
+      free_x <- isTRUE(b$layout$facet$params$free$x)
```

Comment:
Instead accessing `b` directly, we should always access the `layout` object which was returned by `summarise_layout()` above. We'll probably need a change to ggplot2 to get the information that we want.

Reply:
In light of the fact that I've know realized we need to track this information is even more scenarios, I think we should always track the `discrete_mapping` which removes the need to have to check for free scales.